Coordinate.cpp

```diff
@@ -1,11 +1,78 @@
 #include "Coordinate.h"
+#include <cmath>
 
 Coordinate::Coordinate() : x(0), y(0) {
 }
 
 Coordinate::Coordinate(int x, int y) : x(x), y(y) {
 }
 
+Coordinate::Coordinate(Direction d)
+{
+	switch (d) {
+		case N:
+			*this = Coordinate(0, 1);
+			break;
+		case S:
+			*this = Coordinate(0, -1);
+			break;
+		case W:
+			*this = Coordinate(-1, 0);
+			break;
+		case E:
+			*this = Coordinate(1, 0);
+			break;
+		case NE:
+			*this = Coordinate(1, 1);
+			break;
+		case NW:
+			*this = Coordinate(-1, 1);
+			break;
+		case SE:
+			*this = Coordinate(1, -1);
+			break;
+		case SW:
+			*this = Coordinate(-1, -1);
+			break;
+		default:
+			break; // will never reach this state
```

Comment:
Actually it does reach that state.
```
NodeCoordinate n = NodeCordinate( x ,y ) + DirectionVector( NONE, 0);
// expect 
//n.x = x
//n.y = y
```

However x and y are uninitialized and on my computer become -858993460